### PR TITLE
Add Details tab page stories and components

### PR DESCRIPTION
**Pull request by kahboom**
This PR is for the Details (default) tab of the Integration Details page.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/596/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/614/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/622/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/651/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/659/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/674/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/678/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/749/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/815/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/846/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/packages/ui/src/Integration/IntegrationDetailHistoryListView.tsx [hunk 1]
```diff
@@ -46,7 +47,7 @@ export class IntegrationDetailHistoryListView extends React.Component<
           </Grid.Row>
         ) : null}
 
-        {this.props.children ? (
+        {this.props.children && this.props.hasHistory === true ? (
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Integration/IntegrationDetailHistoryListView.tsx]: You could just pass the whole child element here and not bother having a `hasHistory`, then in the parent component do:
```
{ hasHistory && <Grid.Row blah blah blah
```
like [this](https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/packages/ui/stories/Integration/CiCd/CiCdManagePageUI.stories.tsx#L179) kinda thing

**Comment by gashcrumb** [comment]
So I think this going the right direction.  I think you could take the 'draft` and `WIP` off and get this building and we can merge this, then you can work on smaller, more focused PRs from now on.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/911/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by kahboom** [comment]
This PR is ready for review again, whenever someone gets the chance. Thanks!

File: app/ui-react/syndesis/src/modules/integrations/pages/detail/DetailsPage.tsx [hunk 2]
```diff
@@ -22,18 +74,237 @@ export interface IDetailsPageState {
...
+                                  ? data.deployments
+                                  : [];
+                                const editAction: IIntegrationAction = {
+                                  href: resolvers.integration.edit.index({
+                                    flowId: '0',
+                                    integration: data,
+                                  }),
+                                  label: 'Edit',
+                                };
+                                const startAction: IIntegrationAction = {
```
Comment (riccardo-forina) [inline app/ui-react/syndesis/src/modules/integrations/pages/detail/DetailsPage.tsx]: Maybe open an issue to refactor this in a later moment? I really don't like how this is repeated somewhere else in the codebase.

**Comment by kahboom** [comment]
Made improvements based on feedback in this PR: https://github.com/syndesisio/syndesis-react/pull/203

**Pull request merged**